File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/http/client/bind/type/MetadataClient.kt
package io.micronaut.docs.http.client.bind.type

import io.micronaut.http.annotation.Get
import io.micronaut.http.client.annotation.Client

//tag::clazz[]
@Client("/")
interface MetadataClient {
    @Get("/client/bind")
    operator fun get(metadata: Metadata?): String?
}
//end::clazz[]

File: test-suite-python/src/test/python/micronaut/docs/http/client/bind/type/MetadataClient.py
from abc import ABC, abstractmethod

from .Metadata import Metadata

from micronaut.http.annotation import Get
from micronaut.http.client.annotation import Client


# tag::clazz[]
@Client("/")
class MetadataClient(ABC):

    @Get("/client/bind")
    @abstractmethod
    def get(self, metadata: Metadata) -> str:
        ...
# end::clazz[]
